lib/ms2ex/fields/helper.ex

```diff
@@ -10,30 +19,32 @@ defmodule Ms2ex.FieldHelper do
 
     Logger.info("Field #{state.field_id} @ Channel #{state.channel_id}: #{character.name} joined")
 
-    character_ids = Map.keys(state.sessions)
-
     # Load other characters
-    for {_id, char} <- World.get_characters(character_ids) do
-      send(session_pid, {:push, Packets.FieldAddUser.bytes(char)})
-      send(session_pid, {:push, Packets.ProxyGameObj.load_player(char)})
-
-      if mount = Map.get(state.mounts, char.id) do
-        send(session_pid, {:push, Packets.ResponseRide.start_ride(char, mount)})
+    for char_id <- Map.keys(state.sessions) do
+      with {:ok, char} <- CharacterManager.lookup(char_id) do
+        send(session_pid, {:push, Packets.FieldAddUser.bytes(char)})
+        send(session_pid, {:push, Packets.ProxyGameObj.load_player(char)})
+
+        if mount = Map.get(state.mounts, char.id) do
+          send(session_pid, {:push, Packets.ResponseRide.start_ride(char, mount)})
+        end
       end
     end
 
     # Update registry
-    character = %{character | object_id: state.counter, map_id: state.field_id}
+    character = %{character | object_id: state.counter, field_id: state.field_id}
     character = Map.put(character, :field_pid, self())
-    World.update_character(character)
+    CharacterManager.update(character)
 
     sessions = Map.put(state.sessions, character.id, session_pid)
     state = %{state | counter: state.counter + 1, sessions: sessions}
 
     # Load Mobs
-    for {_id, mob} <- state.mobs do
-      send(session_pid, {:push, Packets.FieldAddNpc.add_mob(mob)})
-      send(session_pid, {:push, Packets.ProxyGameObj.load_npc(mob)})
+    for {obj_id, _pid} <- state.mobs do
+      with {:ok, mob} <- Mob.lookup(character, obj_id) do
+        send(session_pid, {:push, Packets.FieldAddNpc.add_mob(mob)})
+        send(session_pid, {:push, Packets.ProxyGameObj.load_npc(mob)})
```

Comment:
this is repeated a lot, maybe creating a util module for pushing these messages?

Reply:
Yep, i still have to rework that code. It's old.